library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -480,7 +480,11 @@ public void updateStatus(int status) {
     }
 
     private boolean isClientReadyToDownload(DownloadBatch downloadBatch) {
-        return downloadClientReadyChecker.isAllowedToDownload(downloadBatch);
+        return downloadClientReadyChecker.isAllowedToDownload(marshallToDownload(downloadBatch));
```

Comment:
we could have a marshaller class and call it `PublicFacingDownloadMarshaller` that'd clear things up (and improve testability)